### DISCUSSION TITLE
clarification on PorousFlowAqueousPreDisChemistry

### DISCUSSION THREAD

**Discussion by navier-rgb** [idaholab/moose#26863] [Q&A Modules: Porous Flow]
Hello,
Could anyone please offer clarification on the following lines of code from `PorousFlowAqueousPreDisChemistry`? 

https://github.com/idaholab/moose/blob/838e1c63f9aa3f6520da6639a2c8842eaf5366c3/modules/porous_flow/src/materials/PorousFlowAqueousPreDisChemistry.C#L350C4-L368C6 

How did PorousFlow compute the "bounded" `_reaction_rate[_qp][r]`? Is it a derivative of some equation? Thank you.

**Comment by navier-rgb** [comment]
@GiudGiud @cpgr

**Reply by GiudGiud** [reply]
It s computed in that material. Seemingly the rate is simply back-computed from the change in another quantity. This expression also implicity assumes a 1st order scheme

**Reply by cpgr** [reply]
I think it does a logic check to see if the concentration after the reaction would be outside the physical range [0,1], given the timestep size and computed reaction rate.  It the back-calculates a reaction rate to limit the concentration to these bounds, as @GiudGiud said.
